vpc/service/assign_addresses_v3.go

```diff
@@ -719,21 +749,25 @@ func (vpcService *vpcService) assignIPsToENI(ctx context.Context, req *vpcapi.As
 		_ = tx.Rollback()
 	}()
 
-	err = lockAssignment(ctx, tx, ass.assignmentID)
+	_, err = lockAssignment(ctx, tx, ass.assignmentID)
```

Comment:
It's weird to me that `lockAssignment` now returns and assignment, but we don't care about it?
Unless it is a transition assignment, then we do? What is up with that?

Reply:
lock assignment always queried the assignment. It also keeps a pessimistic lock on it for non-serializable transactions. This prevents two people from trying to assign IPs to it. It returns a "special" form of assignment (in case there's a race here). I don't really want to create *another* function that does the same thing (locks the assignment), so it seems like a good place to do two birds with one stone.